components/Tables/ContributorsTable.tsx

```diff
@@ -0,0 +1,340 @@
+                    />
+                    <p>{info.row.original.login}</p>
+                  </HoverCard.Trigger>
+                </Link>
+                <HoverCard.Portal>
+                  <HoverCard.Content sideOffset={5}>
+                    <HoverCardWrapper username={info.row.original.login} />
+                  </HoverCard.Content>
+                </HoverCard.Portal>
+              </HoverCard.Root>
+            </div>
+          ),
+        }),
+        contributorsColumnHelper.accessor("oscr", {
+          header: "Rating",
+          enableSorting: true,
+          cell: (info) => <OscrPill rating={info.row.original.oscr ?? 0} />,
+        }),
+        {
+          id: "expand",
+          header: "",
+          cell: ({ row }: { row: Row<DbRepoContributor> }) => {
+            return (
+              row.getCanExpand() && (
+                <button onClick={row.getToggleExpandedHandler()}>
+                  {row.getIsExpanded() ? <BsArrowsCollapse /> : <BsArrowsExpand />}
+                </button>
+              )
+            );
+          },
+        },
+      ],
+    },
+  ];
+
+  const table = useReactTable({
+    columns: useMemo(() => (isMobile ? mobileColumns : defaultColumns), [isMobile]),
+    data: contributors ?? [],
+    manualSorting: true,
+    manualPagination: true,
+    getCoreRowModel: getCoreRowModel(),
+    getRowCanExpand: () => isMobile,
+    onRowSelectionChange: setSelectedContributors,
+    getExpandedRowModel: getExpandedRowModel(),
+    getRowId: (row) => row.login,
+    state: { sorting, rowSelection: selectedContributors },
+  });
+
+  return (
+    <div className="flex flex-col gap-4">
+      {Object.keys(selectedContributors).length > 0 && (
+        <div className="flex justify-between">
+          <p>{Object.keys(selectedContributors).length} selected</p>
+          {isMobile ? (
+            <AddToContributorInsightDrawer repository={repository} contributors={Object.keys(selectedContributors)} />
+          ) : (
+            <Button variant="primary" onClick={() => setIsAddToContributorInsightModalOpen(true)}>
+              Add to Insight
+            </Button>
+          )}
+        </div>
+      )}
+
+      <Table>
+        <TableHeader>
+          {table.getHeaderGroups().map((headerGroup) => (
+            <TableRow key={headerGroup.id}>
+              {headerGroup.headers.map((header) => (
+                <TableHead key={header.id} className="font-semibold">
+                  {header.column.columnDef.enableSorting ? (
+                    <button
+                      onClick={() => {
+                        const { enableSorting } = header.column.columnDef;
+                        const isAscending = Boolean(sorting.find((item) => item.id === header.id && !item.desc));
+
+                        if (enableSorting) {
+                          setOscrSorting(isAscending ? "ASC" : "DESC");
+
+                          setSorting((currentState) => {
```

Comment:
TanStack Table can handle all of this logic if you want with the `header.column.getToggleSortingHandler()` API or `column.toggleSorting`

Then there are table and column def options like `sortDescFirst`, `enableMultiSort`, `enableSortingRemoval`, `maxMultiSortColCount`, etc. to customize the behavior.